packages/react-heartwood-components/src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,253 @@
+						: prevState.highlightedActionIndex - 1
+			}))
+		}
+
+		// Escape
+		if (e.key === 'Escape') {
+			this.setState(
+				{
+					isVisible: false,
+					highlightedActionIndex: -1
+				},
+				() => this.manageListeners()
+			)
+		}
+
+		// Enter
+		if (e.keyCode === 13) {
+			// Check if an action is highlighted
+			if (highlightedActionIndex > -1) {
+				// Trigger it if so
+				if (actions[highlightedActionIndex].onClick) {
+					actions[highlightedActionIndex].onClick()
+				}
+				this.toggleActionsVisibility()
+			}
+		}
+	}
+
+	public manageListeners = () => {
+		if (typeof document !== 'undefined') {
+			if (this.state.isVisible) {
+				document.addEventListener('click', this.handleClickOutside, false)
+				document.addEventListener('keyup', this.onKeyUp, false)
+			} else {
+				document.removeEventListener('click', this.handleClickOutside, false)
+				document.removeEventListener('keyup', this.onKeyUp, false)
+			}
+		}
+	}
+
+	public getButtonPosition = () => {
+		const buttonPosition =
+			this.ref.current && this.ref.current.getBoundingClientRect()
+
+		if (buttonPosition) {
+			return buttonPosition
+		}
+
+		return null
+	}
+
+	public getMenuPosition = () => {
+		const buttonPosition = this.getButtonPosition()
+		const scrollTop =
+			typeof document !== 'undefined' &&
+			document &&
+			document.documentElement &&
+			document.documentElement.scrollTop
+				? document.documentElement.scrollTop
+				: 0
+
+		if (!buttonPosition) {
+			return
+		}
+		const menuPosition = {
+			left: buttonPosition.left,
+			top: buttonPosition.bottom + scrollTop + 2,
+			width: buttonPosition.width
+		}
+
+		this.setState({
+			menuPosition
+		})
+	}
+
+	public toggleActionsVisibility = () => {
+		this.getMenuPosition()
+		this.setState(
+			prevState => ({
+				isVisible: !prevState.isVisible,
+				highlightedActionIndex: prevState.isVisible
+					? -1
+					: prevState.highlightedActionIndex
+			}),
+			() => this.manageListeners()
+		)
+	}
+
+	public render(): React.ReactElement {
+		const { defaultAction, actions, kind, isFullWidth, isSmall } = this.props
+		const { isVisible, menuPosition, highlightedActionIndex } = this.state
+
+		if (!actions || (actions && actions.length === 0)) {
+			// TODO: Warn dev if not in production environment; they might wanna use a different component
+			return (
+				<Button
+					kind={kind}
+					isFullWidth={isFullWidth}
+					isSmall={isSmall}
+					{...defaultAction}
+				/>
+			)
+		}
+
+		return (
+			<div
+				className={cx('split-button', {
+					'split-button--is-full-width': isFullWidth
+				})}
+				ref={this.ref}
+			>
+				<Button
+					kind={kind}
+					isSmall={isSmall}
+					className="split-button__default"
+					{...defaultAction}
+				/>
+				<Button
+					kind={kind}
+					isSmall={isSmall}
+					className="split-button__actions"
+					icon={{ name: 'keyboard_arrow_down' }}
+					onClick={this.toggleActionsVisibility}
+				/>
+				{typeof document !== 'undefined' &&
+					document &&
```

Comment:
I don't **think** you need to check `document` since you verified it wasn't undefined on the previous line

Reply:
Got it. Took those out.